crates/starknet-server/src/cli.rs

```diff
@@ -93,6 +105,12 @@ impl Args {
                 "TESTNET2" => StarknetChainId::TestNet2,
                 _ => panic!("Invalid value for chain-id"),
             },
+            dump_on: match self.dump_on.clone().unwrap_or_default().as_str() {
```

Comment:
If dump_on specified shalln't we check that `dump_path != None`? What about other way around? @FabijanC

Reply:
In devnet-py, there is this check:
```python
    if parsed_args.dump_on and not parsed_args.dump_path:
        sys.exit("Error: --dump-path required if --dump-on present")
```

Reply:
What if `dump_path` specified shall be dump_on specified as well?

Reply:
In devnet-py, there are two modes of dumping:
1. dumping on user's HTTP request
2. dumping on an event:
    a. after each tx
    b. on devnet exit

In that sense, providing only dump_path is ok for devnet-py, because that means the user will dump via HTTP endpoint. But I haven't reviewed this PR yet, so I don't know if the same logic applies here.

Reply:
Fixed in 886ddf0bdaef2dd5e9a2470b1e24d782e2baf6fe and 855d916694823aa429a1c65bde11740e9dc4f7ff

Reply:
Marking this conversation as resolved, but I guess we'll deal with these unfriendly panics in the same PR that will resolve #34.